src/types/decimal.rs

```diff
@@ -296,8 +296,42 @@ impl TryFrom<f64> for Decimal {
 
 impl Display for Decimal {
     fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
-        // TODO: This is correct, but not the most human-friendly format.
-        write!(f, "{}d{}", self.coefficient, self.exponent)
+        let digits = &*self.coefficient.magnitude.to_string();
+        let len = digits.len();
+        // The index of the decimal, relative to the magnitude representation
+        //       0123
+        // Given ABCDd-2, the decimal gets inserted at position 2.
+        let i_d = len as i64 + self.exponent;
+
+        if self.coefficient.sign == Sign::Negative {
+            write!(f, "-").unwrap();
+        };
+
+        if self.exponent == 0 && len > 6 {
```

Comment:
Could you make `6` a named constant and/or add a comment explaining how you chose it?

Reply:
Will do both. Happy to change to some other value, as well.